### ISSUE TITLE
local pdb file cannot be visualized

### ISSUE THREAD

**Issue by enriquead1** [nvelden/NGLVieweR#13]
I am running the most simple script, but I get only a black screen. The current version of the package is 1.3.1 in my R session and I am running it from Ubuntu 22.04.

```r
library(NGLVieweR)

#Load local pdb file
NGLVieweR("7CID.pdb") %>%
  addRepresentation("cartoon")
``` 

Rstudio version is:
RStudio 2023.03.1+446 "Cherry Blossom" Release (6e31ffc3ef2a1f81d377eeccab71ddc11cfbd29e, 2023-05-09) for Ubuntu Jammy
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) rstudio/2023.03.1+446 Chrome/108.0.5359.179 Electron/22.0.3 Safari/537.36

**Comment by nvelden** [comment]
Are you sure you have saved the .pdb file in your working directory? As a test you can visualize the protein directly by just entering the pdb code: 

```
NGLVieweR("7CID") %>%
  addRepresentation("cartoon")
```

**Comment by enriquead1** [comment]
Yes, I am sure. I will upload a short screen recording. Your example works, but mine does not...

[Capture vidéo du 24-07-24 00:08:52.webm](https://github.com/user-attachments/assets/14dda013-8b7b-462d-a055-f49cf06b3bb6)

**Comment by nvelden** [comment]
File names are case sensitive. Could you try `7cid.pdb` instead of `7CID.pdb`?

**Comment by enriquead1** [comment]
I tried, the same keeps happening. I tried clicking on "show in a new window", and it displays the image in the browser, but it still shows only a black screen.

**Comment by nvelden** [comment]
I cannot reproduce your issue. Could you check if the file exists by running `file.exists("7cid.pdb")`. If it returns FALSE, try again using the full file path. To make sure it is not a file format issue could you download the PDB file from the RCSB PDB [website](https://www.rcsb.org/structure/7CID) using Download Files > PDB format. Let me know if that solved the issue.

**Comment by enriquead1** [comment]
I found the problem, it is inside the NGLVieweR function in the version that I am using (1.3.1, from CRAN). The logical operator used is ">" instead of ">=" or even "> 5". As my filename contains just 8 characters, it travels the if statement as a pdb code. At line 27 it turns into "7cid.pdb.pdb".

It works by running this instead:
`data = "7cid.pdb"
format = "pdb"
width = NULL
height = NULL
elementId = NULL


type <- NULL
file_ext <- format
if (file.exists(data) && nchar(data) > 5) {
  if (is.null(format)) {
    file_ext <- tools::file_ext(data)
  }
  data <- paste(readLines(data), collapse = "\n")
  type <- "file"
}else if (nchar(data) > 5 && tools::file_ext(data) == "") {
  type <- "file"
  file_ext <- format
  data <- paste(data, collapse = "\n")
}else {
  type <- "code"
  data <- sprintf("rcsb://%s.pdb", data)
}

x <- list()
x$file_ext <- format
x$data <- data
x$type <- type
x$stageParameters <- list()
x$addRepresentation <- list()
x$addRepresentation$type <- list()
x$addRepresentation$values <- list()
x$setQuality <- "medium"
x$setRock <- FALSE
x$toggleRock <- FALSE
x$setSpin <- FALSE
x$setFocus <- 0
x$toggleSpin <- FALSE
x$zoomMove <- list()
jjj <- htmlwidgets::createWidget(name = "NGLVieweR", x, width = width, 
                          height = height, package = "NGLVieweR", elementId = elementId)

print(type)
print(data)
addRepresentation(jjj, "cartoon")`

**Comment by nvelden** [comment]
Thanks a lot for looking into this. I am a bit confused because this looks like an older version of the function. In the latest version on CRAN the function has been changed to also work with shorter file names. Could you try to remove your version and reinstall?

**Comment by enriquead1** [comment]
I have tried, but the same function is installed. I downloaded "package source" from [CRAN](https://cran.r-project.org/web/packages/NGLVieweR/index.html), looked into the R directory, NGLVieweR.R file, and the code contains "> 8". 

Probably there is some lag on the update of CRAN servers, or something similar. I am unfamiliar with the administration of publicly available R libraries. Could launching a new package version (1.3.2) solve this?
